index.d.ts

```diff
@@ -0,0 +1,2148 @@
+type DeepPartial<T> = {
+  [P in keyof T]?: DeepPartial<T[P]>;
+};
+
+export declare interface IFieldError {
+  /**
+   * Field of the request that the error describes
+   */
+  field: string;
+
+  /**
+   * Human readable description of the problem
+   */
+  message: string;
+}
+
+export declare interface IDatedObject {
+  /**
+   * Date ISO String 
+   */
+  created_at: string;
+
+  /**
+   * Date ISO String 
+   */
+  updated_at: string;
+}
+
+export declare interface IBaseObject<ObjectName> {
+  object: ObjectName;
+}
+
+export declare interface IObjectWithId<ObjectName> extends IBaseObject<ObjectName> {
+  /**
+   * Unique identifier, begins with "adr_" / "prcl_" / "ins_" / "trk_" / "batch_" / "cstinfo_" / etc
+   */
+  id: string;
+
+  /**
+   * Set based on which api-key you used, either "test" or "production"
+   */
+  mode: "test" | "production";
+}
+
+// TODO there seems to be a lot of duplicate codes listed here https://www.easypost.com/errors-guide
+export declare type TErrorCode =
+  "INTERNAL_SERVER_ERROR" | "NOT_ACCEPTABLE" | "NOT_FOUND" | "FORBIDDEN" | "PAYMENT_REQUIRED" | "UNAUTHORIZED" |
+  "BAD_REQUEST" | "PAYMENT_GATEWAY.ERROR" | "MODE.UNAUTHORIZED" | "MODE.CONFLICT" | "DATE.PARSE.FAILURE" | "PARAMETER.FORBIDDEN" |
+  "PARAMETER.REQUIRED" | "ADDRESS.PARAMETERS.INVALID_CHARACTER" | "ADDRESS.PARAMETERS.INVALID" | "ADDRESS.COUNTRY.INVALID" | "ADDRESS.VERIFICATION.NOT_FOUND" | "ADDRESS.VERIFICATION.FAILURE" |
+  "ADDRESS.VERIFY.UNAVAILABLE" | "ADDRESS.VERIFICATION.INVALID" | "ADDRESS.VERIFICATION.NOT_FOUND" | "ADDRESS.VERIFY.FAILURE" | "ADDRESS.VERIFY.CARRIER_INVALID" | "ADDRESS.VERIFY.UPSTREAM_UNAVAILABLE" |
+  "ADDRESS.VERIFY.ONLY_US" | "ADDRESS.VERIFY.INTL_NOT_ENABLED" | "ADDRESS.VERIFY.MISSING_STREET" | "ADDRESS.VERIFY.MISSING_CITY_STATE_ZIP" |
+
+  // TODO typo "FORAMT" should be "FORMAT". Is the typo just in the docs or in the api error response as well?
+  "BATCH.FILE_FORAMT.INVALID" |
```

Comment:
Error code ["BATCH.FILE_FORAMT.INVALID"](https://www.easypost.com/errors-guide#error-codes) has a typo "FORAMT" - should be "FORMAT". 
Is the typo just in the docs or in the api error response as well?

Reply:
Nice catch! This is unfortunately incorrect in both the documentation as well as what gets returned from the API. I've brought it up with our team; however, because it's been misspelled for some time, it may stay incorrect for the foreseeable future as some integrations may have coded against this typo. We have potential plans of versioning our API in which case this could be corrected; however, these plans will take some time.